docs: Cluster peering 1.14 beta release (#14794)
hashicorp/consul: website/content/docs/connect/cluster-peering/k8s.mdx

- `PeeringAcceptor`: Generates a peering token and accepts an incoming peering connection.
- `PeeringDialer`: Uses a peering token to make an outbound peering connection with the cluster that generated the token.

As of Consul v1.14, you can also [implement service failovers and redirects to control traffic](/consul/docs/connect/l7-traffic) between peers.

## Prerequisites

You must implement the following requirements to create and use cluster peering connections with Kubernetes:

- Consul v1.13.1 or later
- At least two Kubernetes clusters
- The installation must be running on Consul on Kubernetes version 0.47.1 or later

### Prepare for installation

> 1. After you provision a Kubernetes cluster and set up your kubeconfig file to manage access to multiple Kubernetes clusters, use the `kubectl` command to export the Kubernetes context names and then set them to variables for future use. For more information on how to use kubeconfig and contexts, refer to the [Kubernetes docs on configuring access to multiple clusters](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/).

Reviewer: ```suggestion
Complete the following procedure after you have provisioned a Kubernetes cluster and set up your kubeconfig file to manage access to multiple Kubernetes clusters. 

1. Use the `kubectl` command to export the Kubernetes context names and then set them to variables for future use. For more information on how to use kubeconfig and contexts, refer to the [Kubernetes docs on configuring access to multiple clusters](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/).
```
The clause at the beginning of the step might make more sense as an introduction to the process. Splitting it out might make it easier for readers to consume the information.

    $ helm install ${HELM_RELEASE_NAME} hashicorp/consul --create-namespace --namespace consul --version "0.47.1" --values values.yaml --kube-context $CLUSTER2_CONTEXT
     ```

> ## Create a peering connection for Consul on Kubernetes

Reviewer: ```suggestion
## Create a peering connection for Consul on Kubernetes

To peer Kubernetes clusters running Consul, you need to create a peering token and share it with the other cluster. Complete the following steps to create the peer connection.
```
Is this an acceptable way to introduce this group of procedures?

### Create a peering token

> To peer Kubernetes clusters running Consul, you need to create a peering token and share it with the other cluster. Peers identify each other using the `metadata.name` values you establish when creating the `PeeringAcceptor` and `PeeringDialer` CRDs.

Reviewer: ```suggestion
Peers identify each other using the `metadata.name` values you establish when creating the `PeeringAcceptor` and `PeeringDialer` CRDs.
```
Moved to the paragraph introducing the group of procedures

    $ kubectl --context $CLUSTER2_CONTEXT apply --filename dialer.yaml 
    ```

### Export services between clusters

> ~> **Tip**: The examples in this section demonstrate how to export a service named `backend`. You should change instances of `backend` in the example code to the name of the service you want to export.

Reviewer: ```suggestion
The examples described in this section demonstrate how to export a service named `backend`. You should change instances of `backend` in the example code to the name of the service you want to export.
```
I wouldn't characterize this information as a tip. It's salient information for understanding the code snippets

    </CodeBlockConfig>

> 2. After you update `PeeringAcceptor`, repeat the steps to create a peering connection, including saving a new peering token and exporting it to the other cluster. Your peering connection is re-established with the updated token.

Reviewer: ```suggestion
2. After updating `PeeringAcceptor`, repeat the following steps to create a peering connection:
      1.  [Create a peering token](#create-a-peering-token)
      1.  [Establish a peering connection between clusters](#establish-a-peering-connection-between-clusters)
      1. [Export services between clusters](#export-services-between-clusters)
      1. [Authorize services for peers](#authorize-services-for-peers)
      
      Your peering connection is re-established with the updated token.
```
Correct? These are the steps they need to repeat?

~> **Note:** The only way to create or set a new peering token is to manually adjust the value of the annotation `consul.hashicorp.com/peering-version`. Creating a new token causes the previous token to expire.

## Traffic management between peers

> As of Consul v1.14, you can use [dynamic traffic management](/consul/docs/connect/l7-traffic) to configure your service mesh so that services automatically failover and redirect between peers.

Reviewer: ```suggestion
Use [dynamic traffic management](/consul/docs/connect/l7-traffic) to configure your service mesh so that services automatically failover and redirect between peers.
```
Same comment about versioning
